src/main/common/components/payment-intention/payment-date.ts

```diff
@@ -0,0 +1,77 @@
+import * as express from 'express'
+import { Moment } from 'moment'
+
+import { AbstractModelAccessor } from 'shared/components/model-accessor'
+import { PaymentIntention } from 'shared/components/payment-intention/model/paymentIntention'
+import { PaymentType } from 'shared/components/payment-intention/model/paymentOption'
+import { PaymentDate } from 'shared/components/payment-intention/model/paymentDate'
+import { Paths } from 'shared/components/payment-intention/paths'
+
+import { GuardFactory } from 'response/guards/guardFactory'
+import { ErrorHandling } from 'shared/errorHandling'
+import { NotFoundError } from 'errors'
+import { MomentFactory } from 'shared/momentFactory'
+
+import { User } from 'idam/user'
+import { Form } from 'forms/form'
+import { FormValidator } from 'forms/validation/formValidator'
+import { DraftService } from 'services/draftService'
+
+export abstract class AbstractPaymentDatePage<Draft> {
+
+  abstract getHeading (): string
+  abstract createModelAccessor (): AbstractModelAccessor<Draft, PaymentIntention>
+  abstract buildTaskListUri (req: express.Request, res: express.Response): string
+
+  getView (): string {
+    return 'components/payment-intention/payment-date'
+  }
+
+  buildRouter (path: string, ...guards: express.RequestHandler[]): express.Router {
+    const stateGuardRequestHandler: express.RequestHandler = GuardFactory.create((res: express.Response): boolean => {
+      const model: PaymentIntention = this.createModelAccessor().get(res.locals.draft.document)
+
+      return model && model.paymentOption && model.paymentOption.isOfType(PaymentType.BY_SET_DATE)
+    }, (req: express.Request): void => {
+      throw new NotFoundError(req.path)
+    })
+
+    return express.Router()
+      .get(
+        path + Paths.paymentDatePage.uri,
+        ...guards,
+        stateGuardRequestHandler,
+        (req: express.Request, res: express.Response) => {
+          this.renderView(new Form(this.createModelAccessor().get(res.locals.draft.document).paymentDate), res)
+        })
+      .post(
+        path + Paths.paymentDatePage.uri,
+        ...guards,
+        stateGuardRequestHandler,
+        FormValidator.requestHandler(PaymentDate, PaymentDate.fromObject),
+        ErrorHandling.apply(async (req: express.Request, res: express.Response) => {
+          const form: Form<PaymentDate> = req.body
+
+          if (form.hasErrors()) {
+            this.renderView(form, res)
+          } else {
+            this.createModelAccessor().patch(res.locals.draft.document, model => model.paymentDate = form.model)
+
+            const user: User = res.locals.user
+            await new DraftService().save(res.locals.draft, user.bearerToken)
+
+            res.redirect(this.buildTaskListUri(req, res))
+          }
+        }))
+  }
+
+  private renderView (form: Form<PaymentDate>, res: express.Response) {
+    const futureDate: Moment = MomentFactory.currentDate().add(1, 'month')
```

Comment:
We need to be consistent with how we do example dates please update to use similar logs as this example: 
https://github.com/hmcts/cmc-citizen-frontend/pull/804/files#diff-f513f2f23d4687729ac795343d516a45R84

Reply:
Good spot, reimplemented to use Andrew's approach.